soh/soh/Enhancements/randomizer/randomizer.cpp

```diff
@@ -440,6 +441,116 @@ void Randomizer::LoadMerchantMessages(const char* spoilerFileName) {
             "%gobjet mystérieux%w pour 60 Rubis?\x1B&%gOui&Non%w",
         });
 
+    //Setup for merchant text boxes
+    //Medigoron
+    //RANDOTODO: Implement obscure/ambiguous hints
+    RandomizerGet medigoronGet = this->itemLocations[RC_GC_MEDIGORON].rgID;
+    std::vector<std::string> mgItemName;
+    if (Randomizer::GetRandoSettingValue(RSK_SHUFFLE_MERCHANTS) == 2) {//If merchant hints are clear... 
+        if (medigoronGet == RG_ICE_TRAP) {
+            medigoronGet = this->itemLocations[RC_GC_MEDIGORON].fakeRgID;
+            mgItemName = {
+                this->itemLocations[RC_GC_MEDIGORON].trickName, 
+                this->itemLocations[RC_GC_MEDIGORON].trickName,
+                this->itemLocations[RC_GC_MEDIGORON].trickName
+            };
+        } else {
+            mgItemName = EnumToSpoilerfileGetName[medigoronGet];
+        }
+    } else {
+        mgItemName = {
+            "something cool",
+            "etwas cooles",
+            "un truc cool"
+        };
+    }
+         
+    CustomMessageManager::Instance->CreateMessage(
+        Randomizer::merchantMessageTableID, TEXT_MEDIGORON,
+        {
+            TEXTBOX_TYPE_BLACK,
+            TEXTBOX_POS_BOTTOM,
+            "How about buying %r&" + mgItemName[0] + "%w for %g200 rupees%w?\x1B&%gYes&No%w",
+            "Wie wäre es mit %r&" + mgItemName[1] + "%w für %g200 Rubine?%w\x1B&%gJa!&Nein!%w",
+            "Qu'est-ce que tu dirais de l'acheter pour%g200 Rubis?%w\x1B&%gOui&Non"
```

Comment:
How come the item name isn't added for french?

Reply:
I spoke with @PurpleHato on this and it had something to do with the gender of the article preceding the item. I couldn't fond a suitable replacement which included the item for French.

Reply:
Hmm, I obviously don't know enough about french to understand but I'd opt for showing the item name in an awkward way that doesn't always make 100% sense over being inconsistent and lacking functionality (this would quite literally lead to an advantage of using non french in this case)

Worth mentioning even in english sometimes dynamically built strings don't always make sense either, eg: 
"I'll sell you a Odd Potion"
"For 1 rupees"

Reply:
Actually, it seems like 3D rando might have a pre-baked option which allows for insertion of the check name, so I might look into bringing that over.

Reply:
> Hmm, I obviously don't know enough about french to understand but I'd opt for showing the item name in an awkward way that doesn't always make 100% sense over being inconsistent and lacking functionality (this would quite literally lead to an advantage of using non french in this case)
> 
> Worth mentioning even in english sometimes dynamically built strings don't always end up perfect either, eg: "I'll sell you a Odd Potion" "For 1 rupees"

I'll document what will need to be what but that gonna take me time and no idea how we'll want to manage that anyway

Reply:
After searching a little bit, we can call the clear hint text from the hint_list_table.cpp, since it contains everything we want for each language @garrettjoecox @stratomaster64 
![image](https://user-images.githubusercontent.com/47987542/198225664-bc8d63f1-cdf6-4583-910c-eaef09a007a5.png)

Reply:
> After searching a little bit, we can call the clear hint text from the hint_list_table.cpp, since it contains everything we want for each language @garrettjoecox @stratomaster64 ![image](https://user-images.githubusercontent.com/47987542/198225664-bc8d63f1-cdf6-4583-910c-eaef09a007a5.png)

Reply:
> After searching a little bit, we can call the clear hint text from the hint_list_table.cpp, since it contains everything we want for each language and include the garman one to them @garrettjoecox @stratomaster64 ![image](https://user-images.githubusercontent.com/47987542/198225664-bc8d63f1-cdf6-4583-910c-eaef09a007a5.png)

Reply:
Ehh, this is a bigger change that we’ll want to cross over into other merchants, I’m not sure if we want this tangled with this PR

Reply:
I agree on the fact that is a big thing, but that's also a necessity to have something properly worky and not having an half-done support

Reply:
I’m not sure translation not being 100% correct would be considered half-done, we’re just missing translations all together for German in a multitude of places. Maybe the correct solution here is just to leave it in English until we can eventually pull in the obscure/ambiguous/clear hint names in a later PR

Reply:
> I’m not sure translation not being 100% correct would be considered half-done, we’re just missing translations all together for German in a multitude of places. Maybe the correct solution here is just to leave it in English until we can eventually pull in the obscure/ambiguous/clear hint names in a later PR

As long as it is a follow up, sure
Also looks like generating a french rando  save file on that specific build is crashing for me, here's my rando json file, can you confirm it's not on my side only ?
[shipofharkinian.zip](https://github.com/HarbourMasters/Shipwright/files/9879455/shipofharkinian.zip)

Reply:
the crash on creating a rando save (especially when using french) is probably related to #1846